test/Microsoft.AspNetCore.Mvc.Core.Test/Routing/UrlHelperWithLinkGeneratorTest.cs

```diff
@@ -0,0 +1,1717 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matchers;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.ObjectPool;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.Routing
+{
+    public class UrlHelperWithLinkGeneratorTest
```

Comment:
Tests which were copied from `UrlHelperTest.cs` and updated to use LG

Reply:
We want to share these tests if possible. Could reuse @rynowak's technique in routing for testing the different route implementations.

Reply:
Sure 👍